cmd/chart-repo/utils_test.go

```diff
@@ -285,16 +285,29 @@ func Test_parseRepoIndex(t *testing.T) {
 func Test_chartsFromIndex(t *testing.T) {
 	r := repo{Name: "test", URL: "http://testrepo.com"}
 	index, _ := parseRepoIndex([]byte(validRepoIndexYAML))
-	charts := chartsFromIndex(index, r)
+	charts := chartsFromIndex(index, r, new(filters))
 	assert.Equal(t, len(charts), 2, "number of charts")
+	// filter on name
+	filter := new(filters)
+	filter.Names = append(filter.Names, "wordpress")
+	filter.Names = append(filter.Names, "not-found")
+    charts = chartsFromIndex(index, r, filter)
+    assert.Equal(t, len(charts), 1, "number of charts")
+	// filter on annotation
+	filter = new(filters)
+	filter.Annotations = make(map[string]string)
+	filter.Annotations["sync"] = "true"
+	filter.Annotations["not-found"] = "missing"
+	filter.Annotations["no-value"] = ""
+    assert.Equal(t, len(charts), 1, "number of charts")
```

Comment:
what chart does this test pick up? I can't see any annotations in https://github.com/helm/monocular/blob/master/cmd/chart-repo/testdata/valid-index.yaml?

Reply:
My bad, I was doing some cleanup and the updated test yam l did got get committed.  It will be there in the next push.